test cases/common/16 configure file/meson.build

```diff
@@ -29,13 +29,23 @@ ifile = '@0@/dummy.dat'.format(meson.current_source_dir())
 ofile = '@0@/config2.h'.format(meson.current_build_dir())
 
 check_file = find_program('check_file.py')
+
 # Configure in source root with command and absolute paths
 configure_file(input : 'dummy.dat',
   output : 'config2.h',
   command : [genprog, scriptfile,  ifile, ofile],
   install_dir : 'share/appdir')
 run_command(check_file, join_paths(meson.current_build_dir(), 'config2.h'))
 
+# Same again, but an input file should not be required in this
+# case where we use a command to generate the output file.
+ofile2b = '@0@/config2b.h'.format(meson.current_build_dir())
+configure_file(
+  output : 'config2b.h',
+  command : [genprog, scriptfile, ifile, ofile2b],
+  install_dir : 'share/appdir')
```

Comment:
It is confusing that the comment says that an input file is not needed but still it is given. There should be a different script that actually does not take inputs or the generator script should be changed so that it can take either 2 or 3 arguments.